padertorch/data/fragmenter.py

```diff
@@ -107,3 +119,190 @@ def fragment(key, x):
             fragment = deflatten(fragment)
             fragments.append(fragment)
         return fragments
+
+
+def get_anchor(
+        num_samples: int, length: int, shift: int=None,
+        mode: str = 'left', rng=np.random
+):
+    """
+    Calculates anchor for the boundaries for segmentation of a signal
+    with length `num_sammples` in case of a fixed segment length
+    `length` and shift `shift`
+
+    Args:
+        num_samples: num samples of signal for which boundaries are caclulated
+        length: segment length
+        shift: shift between segments, defaults to length
+        mode: Defines the position of the boundaries in the signal:
+            left: anchor is set to zero
+                so that only values at the end are cut
+            right: anchor is set to `num_samples`
+                so that only values at the beginning are cut
+            center: anchor is set to `num_samples // 2`
+            centered_cutout: the anchor is chosen such that the same number
+                of samples are discarded at the end and the beginning
+            random: anchor is set to a random value between
+                `0` and `num_samples`.
+                 This may reduce the number of possible segments.
+            random_max_segments: Randomly chooses the anchor such that
+                the maximum number of segments are created
+        rng: random number generator (`numpy.random`)
+
+    Returns:
+       integer value describing the anchor
+    >>> np.random.seed(3)
+    >>> get_anchor(24, 10, 3, mode='left')
+    0
+    >>> get_anchor(24, 10, 3, mode='right')
+    14
+    >>> get_anchor(24, 10, 3, mode='center')
+    12
+    >>> get_anchor(24, 10, 3, mode='centered_cutout')
+    1
+    >>> get_anchor(24, 10, 3, mode='random')
+    10
+    >>> get_anchor(24, 10, 3, mode='random_max_segments')
+    3
+    """
+    assert num_samples >= length, (num_samples, length)
+    if shift is None:
+        shift = length
+    assert shift > 0, shift
+
+    if mode == 'left':
+        return 0
+    elif mode == 'right':
+        return num_samples - length
+    elif mode == 'center':
+        return num_samples // 2
+    elif mode == 'centered_cutout':
+        remainder = (num_samples - length) % shift
+        return remainder // 2
+    elif mode == 'random':
+        return rng.randint(num_samples - length)
```

Comment:
Why `num_samples - length`? Shouldn't this valu be in the range `[0, ..., num_samples]`?
Note I included the last value.